### ISSUE TITLE
QR code scanner

### ISSUE THREAD

**Issue by rugk** [rugk/offline-qr-code#52]
There are also many JS libraries for just _scanning_ a QR code:

* https://github.com/schmich/instascan
* https://serratus.github.io/quaggaJS/
* https://github.com/dwa012/html5-qrcode
* https://github.com/edi9999/jsqrcode
* https://github.com/schmich/instascan

**Comment by rugk** [comment]
This add-on shows what I mean: https://addons.mozilla.org/de/firefox/addon/qrcode-generator-and-reader/

**Comment by rugk** [comment]
Possibly a good lib: https://github.com/zxing-js/library

**Comment by jamescallumyoung** [comment]
@rugk Saw this after your link from #213 

> right-click on image (or canvas, or so) -> decode QR code
I like this. Not too invasive and it's a good contextual option.

> ... scan from webcam
This not so much; I like that this is a small extension. More permissions and camara use would scare me off.

Just my 2 cents.

**Comment by rugk** [comment]
> More permissions and camara use would scare me off.

This would definitively only be requested at runtime when you use it. (It would use WebRTC, which has this built-in.) An add-on actually does not have more permissions than a usual web page in this regards…

**Comment by Harinath333** [comment]
Any one working on this issue???

**Comment by Harinath333** [comment]
If no one I want to work on this using html15-rcode

**Comment by rugk** [comment]
You can do so, that sounds great, but keep in mind this may be a big task and require some back- and forth.

First of, you would have to:
* think of some GUI, like I tried in https://github.com/rugk/offline-qr-code/issues/52#issuecomment-385027154 to get started
* decide which lib to use, which you apparently did, but I'd like to know the reasons of why you'd choose this lib. (Does not mean it'*s wrong, I just like to have it documented or so); also note this issue is old, maybe there are more modern/better libs out there we did not thought about yet – did you have a look for these? My initial listing in the first comment was just meant as a starter for further research, it does not mean one of these libs needs to be used.

**Comment by Harinath333** [comment]
> You can do so, that sounds great, but keep in mind this may be a big task and require some back- and forth.
> 
> First of, you would have to:
> * think of some GUI, like I tried in https://github.com/rugk/offline-qr-code/issues/52#issuecomment-385027154 to get started
> * decide which lib to use, which you apparently did, but I'd like to know the reasons of why you'd choose this lib. (Does not mean it's wrong, I just like to have it documented or so); also note this issue is old, maybe there are more modern/better libs out there we did not thought about yet – did you have a look for these? My initial listing in the first comment was just meant as a starter for further research, it does not mean one of these libs needs to be used.

The

**Comment by Harinath333** [comment]
Hey we just need a qr code scanner from the webcam right! then we can make it by Html5 with qr code lib. Which will let us use the webcam from the user to scan a qr. Let me know if this is what we need

**Comment by rugk** [comment]
Technically it should work with the webcam and yes likely use a lib (which uses the WebRTC technologies for accessing a webcam, I guess)…

But as indicated, one first has to think about a UI and find a good lib. Maybe also look for (preferably open-source) websites that already implement such a feature (and look what they use).

**Comment by Harinath333** [comment]
If no one I want to work on this using html15-rcode

**Comment by Harinath333** [comment]
> Technically it should work with the webcam and yes likely use a lib (which uses the WebRTC technologies for accessing a webcam, I guess)…
> 
> But as indicated, one first has to think about a UI and find a good lib. Maybe also look for (preferably open-source) websites that already implement such a feature (and look what they use).

Yeah I will make one which will have a decent UI and It can use WebRTC

**Comment by klawdhfzasjhaa** [comment]
I'd love that to bridge phone texts / URLs to PC / laptop in some scenarios. :dancers:
